### PR TITLE
MULE-13459: Add the concept of Stereotype to connection providers

### PR DESCRIPTION
**Pull request by alegmarra**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/WithStereotypesDeclaration.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Contract interface for a {@link BaseDeclaration} in which
+ * it's possible to add/get {@link StereotypeModel} objects
+ *
+ * @since 1.0
+ */
+public interface WithStereotypesDeclaration {
+
+  StereotypeModel getStereotype();
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/WithStereotypesDeclaration.java]: Optional

File: src/main/java/org/mule/runtime/api/meta/model/config/ConfigurationModel.java [hunk 2]
```diff
@@ -40,6 +41,7 @@
  * @since 1.0
  */
 public interface ConfigurationModel extends NamedObject, DescribedObject, EnrichableModel, ParameterizedModel,
-    HasOperationModels, HasSourceModels, HasConnectionProviderModels, HasDisplayModel, HasExternalLibraries {
+    HasOperationModels, HasSourceModels, HasConnectionProviderModels, HasDisplayModel,
+    HasExternalLibraries, StereotypedModel {
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/config/ConfigurationModel.java]: implement StereotypeModel? Shouldnt't it be an interface like HasStereotypeModel?

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ConfigurationDeclaration.java [hunk 3]
```diff
@@ -133,4 +135,12 @@ public ConfigurationDeclaration addExternalLibrary(ExternalLibraryModel external
     externalLibraryModels.add(externalLibraryModel);
     return this;
   }
+
+  public StereotypeModel getStereotype() {
+    return stereotype;
```
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ConfigurationDeclaration.java]: would be awesome to be able to inherit this

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/WithStereotypesDeclaration.java (hunk 1 above)
Comment (alegmarra) [reply src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/WithStereotypesDeclaration.java]: everything has an stereotype, it is added with a declaration enricher

**Comment by alegmarra** [comment]
approved

**Pull request merged**